packages/utils/telemetry-utils/src/test/childLogger.spec.ts

```diff
@@ -176,4 +183,182 @@ describe("ChildLogger", () => {
 		childLogger2.send({ category: "generic", eventName: "testEvent" });
 		assert(sent, "event should be sent");
 	});
+
+	describe("Sampling", () => {
+		let events: ITelemetryBaseEvent[] = [];
+		function getBaseLoggerWithConfig(
+			configDictionary?: Record<string, ConfigTypes>,
+		): ITelemetryBaseLogger {
+			const logger: ITelemetryBaseLogger = {
+				send(event: ITelemetryBaseEvent): void {
+					events.push(event);
+				},
+			};
+			const configProvider = (
+				settings: Record<string, ConfigTypes>,
+			): IConfigProviderBase => ({
+				getRawConfig: (name: string): ConfigTypes => settings[name],
+			});
+			return mixinMonitoringContext(logger, configProvider(configDictionary ?? {})).logger;
+		}
+
+		beforeEach(() => {
+			events = [];
+		});
+
+		it("Applies sampling when feature flag to force unsampled telemetry is not set", () => {
+			const eventSamplingConfigs = new Map<
+				string,
+				SystematicSamplingConfig | RandomChanceSamplingConfig
+			>([
+				[
+					"oneEveryTwo",
+					{
+						strategy: SamplingStrategy.SYSTEMATIC,
+						samplingRate: 2,
+					},
+				],
+				[
+					"oneEveryFive",
+					{
+						strategy: SamplingStrategy.SYSTEMATIC,
+						samplingRate: 2,
+					},
+				],
+				[
+					"randomChance",
+					{
+						strategy: SamplingStrategy.RANDOM,
+						percentChance: 0.75,
+					},
+				],
+			]);
+
+			const logger = getBaseLoggerWithConfig();
+			const childLogger = ChildLogger.create(
+				logger,
+				undefined,
+				undefined,
+				eventSamplingConfigs,
+			);
+
+			for (let i = 0; i < 15; i++) {
+				childLogger.send({ category: "generic", eventName: "noSampling" });
+				childLogger.send({ category: "generic", eventName: "oneEveryTwo" });
+				childLogger.send({ category: "generic", eventName: "oneEveryFive" });
+			}
+
+			const totalEmittedRandomChanceEvents = 1000;
+			for (let i = 0; i < totalEmittedRandomChanceEvents; i++) {
+				childLogger.send({ category: "generic", eventName: "randomChance" });
+			}
+
+			// These counts also validate that we issue sampled events the first time we see them, not only until the specified
+			// number of samples have been seen.
+			assert.equal(events.filter((event) => event.eventName === "noSampling").length, 15);
+			assert.equal(events.filter((event) => event.eventName === "oneEveryTwo").length, 8);
+			assert.equal(events.filter((event) => event.eventName === "oneEveryFive").length, 3);
+			assert.equal(
+				events.filter((event) => event.eventName === "randomChance").length >
+					totalEmittedRandomChanceEvents,
+				true,
+			);
+		});
+
+		it("Ignores sampling when feature flag to force unsampled telemetry is set", () => {
```

Comment:
This test we can do deterministically for the random sampling approach too, since we expect no events to be filtered out. Let's add that (in a separate `it()` block), if we keep the current random-sampling approach. If we swap it for something like what id compressor does today the test would probably look pretty different.